packages/@atjson/renderer-react/src/index.ts

```diff
@@ -14,30 +14,24 @@ export type AttributesOf<AnnotationClass> = AnnotationClass extends Annotation<
     }
   : never;
 
-export default class ReactRenderer extends Renderer {
-  private componentLookup: {
-    [key: string]: ComponentType<any>;
-  };
+const ReactRendererContext = React.createContext();
 
-  constructor(
-    document: Document,
-    componentLookup: {
-      [key: string]: ComponentType<any>;
-    }
-  ) {
-    super(document);
-    this.componentLookup = componentLookup;
-  }
+export const ReactRendererConsumer = ReactRendererContext.Consumer;
 
+export const ReactRendererProvider = ({ children, value }) => {
+  return React.createElement(ReactRendererConsumer, {}, parentComponentMap => {
+    const mergedValues = Object.assign({}, parentComponentMap, value);
```

Comment:
We can use the spread operator here instead of `Object.assign`:

```ts
let mergedValues = { ...parentComponentMap, ...value };
```

Reply:
Bad habit, fixed!